filesystem/fat32/dos331bpb.go

```diff
@@ -47,6 +47,37 @@ func dos331BPBFromBytes(b []byte) (*dos331BPB, error) {
 	return &bpb, nil
 }
 
+func (bpb *dos331BPB) clusterCount() uint32 {
+	fatStartSector := bpb.dos20BPB.reservedSectors
+	fatSectors := uint32(bpb.dos20BPB.sectorsPerFat) * uint32(bpb.dos20BPB.fatCount)
+
+	rootDirStartSector := uint32(fatStartSector) + fatSectors
+	rootDirSectors := (uint32(bpb.dos20BPB.rootDirectoryEntries)*32 + uint32(bpb.dos20BPB.bytesPerSector) - 1) / uint32(bpb.dos20BPB.bytesPerSector)
+
+	dataStartSector := rootDirStartSector + rootDirSectors
+	totalSectors := uint32(bpb.dos20BPB.totalSectors)
+	if totalSectors == 0 {
+		totalSectors = bpb.totalSectors
+	}
+	dataSectors := totalSectors - dataStartSector
+
+	return dataSectors / uint32(bpb.dos20BPB.sectorsPerCluster)
+}
+
+// FatType returns the type of FAT filesystem based on the number of clusters
+// http://elm-chan.org/docs/fat_e.html#fat_determination
+func (bpb *dos331BPB) FatType() int {
```

Comment:
I just returned `12`, `16` and `32` here. Happy to create an enum-like type or something else if you think that's better than this.